build-support/travis/generate_travis_yml_main.py

```diff
@@ -0,0 +1,11 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from generate_travis_yml import generate_travis_yml
+
+
+if __name__ == '__main__':
+  generate_travis_yml()
```

Comment:
I'm concerned that the mustache templating adds a significant amount of complexity to what is already a very complicated language (yaml). How would you feel about dropping the mustache templating from this change (ie, allowing the copy-pasting of shards to continue a little while longer), and focusing on using remote-execution/remoting for integration tests as our solution to the copypasta?

The idea with remoting would be to have exactly one integration test shard that would use remoting to execute all integration tests remotely (or one shard for both integration and unit tests, possibly). This requires a bit more investment, but I'm pretty sure it's investment that we need to do before too many more Travis billing cycles have elapsed.

Reply:
I would love to rely on remoting for this, but I would rather eliminate the copypasta now, and get rid of the templates if and when that remoting solution pans out... Fortunately we don't modify .travis.yml that often, and the most common modification (adding shards) is now made easier.

Reply:
Ok. I won't block on it.